src/useCases/redirectTo.ts

```diff
@@ -0,0 +1,16 @@
+import { GetServerSidePropsResult, Redirect } from "next";
+
+export interface IRedirectTo {
+  execute: (url: string) => GetServerSidePropsResult<Redirect>;
+}
+
+export class RedirectTo implements IRedirectTo {
```

Comment:
Maybe a better name, which is inline with the use case of `AuthenticateUser`, would be something along the lines of: `RedirectUserTo` or `RedirectUserToUrl`?